agent/consul/connect_ca_endpoint.go

```diff
@@ -186,8 +187,14 @@ func (s *ConnectCA) ConfigurationSet(
 	if err != nil {
 		return fmt.Errorf("could not initialize provider: %v", err)
 	}
-	if err := newProvider.Configure(args.Config.ClusterID, true,
-		args.Config.Config, args.Config.State); err != nil {
+	pCfg := ca.ProviderConfig{
+		ClusterID:  args.Config.ClusterID,
+		Datacenter: s.srv.config.Datacenter,
+		PrimaryDC:  true,
```

Comment:
Shouldn't this be something like `s.srv.config.Datacenter == s.srv.config.PrimaryDatacenter` since this method can also be used from a secondary to reconfigure just the intermediate?

Reply:
Though if this is a bug, it looks like the `true` was here before your refactor :thinking:

Reply:
Good catch! Yeah I was assuming this was always primary but you are right it's not since each DC CAconfig is independent.

Reply:
Fixed.